components/ee/ws-scheduler/pkg/scheduler/scheduler.go

```diff
@@ -290,22 +307,58 @@ func (s *Scheduler) schedulePod(ctx context.Context, pod *corev1.Pod) (err error
 		return nil
 	}
 
-	// we found a node for this pod - bind it to the node
-	err = s.bindPodToNode(ctx, pod, nodeName)
-	if err != nil {
-		errStr := err.Error()
-		if strings.Contains(errStr, "is already assigned to node") {
-			// some other worker was faster: This can happen and is good: do nothing
-			log.WithFields(owi).WithField("name", pod.Name).Debugf("pod already bound - someone was faster than me!")
-			return nil
+	// if this is a workspace that replaces a ghost: try to find a ghost that is not being targeted, yet
+	var ghostToDelete string
+	if isGhostReplacing {
+		reservedGhosts := s.localSlotCache.getReservedGhostsOnNode(nodeName)
+		ghostToDelete = state.FindOldestGhostOnNodeExcluding(nodeName, reservedGhosts)
+	}
+
+	// mark as already scheduled, even before the actually scheduling has happened.
+	// this enables asynchronous scheduling - we just need to make sure we release the slot in case the scheduling fails!
+	s.localSlotCache.markAsScheduled(pod, nodeName, ghostToDelete)
+
+	// we found a node for this pod: (asynchronously) bind it to the node! schedulingPodMap makes sure we do not try to
+	// select the same slot multiple times.
+	go func(pod *corev1.Pod, ghostToDelete string) {
+		var err error
+		defer s.schedulingPodMap.remove(pod.Name)
+		defer func() {
```

Comment:
nit: could become a single defer. Just for style, with Go 1.15 technically it doesn't matter